dowhy/causal_estimators/causalml.py

```diff
@@ -0,0 +1,108 @@
+import inspect
+import numpy as np
+import pandas as pd
+import pdb
+
+from dowhy.causal_estimator import CausalEstimate
+from dowhy.causal_estimator import CausalEstimator
+from importlib import import_module
+import causalml
+
+class Causalml(CausalEstimator):
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        # Add the core method used in the estimator
+        self.identifier_method = self._target_estimand.identifier_method
+        self.logger.debug("The identifier method used {}".format(self.identifier_method))
+
+        # Check the backdoor variables being used
+        self.logger.debug("Back-door variables used:" +
+                          ",".join(self._target_estimand.backdoor_variables))
+        
+        # Add the observed confounders and one hot encode the categorical variables
+        self._observed_common_causes_names = self._target_estimand.backdoor_variables
+        if self._observed_common_causes_names:
+            # Get the data of the unobserved confounders
+            self._observed_common_causes = self._data[self._observed_common_causes_names]
+            # One hot encode the data if they are cateogorical
+            self._observed_common_causes = pd.get_dummies(self._observed_common_causes, drop_first=True)
+        else:
+            self._observed_common_causes = []
+        
+        # Check the instrumental variables involved
+        self.logger.debug("Instrumental variables used:"+
+                        ",".join(self._target_estimand.instrumental_variables))
+        
+        # Perform the same actions as the above
+        self._instrumental_variable_names = self._target_estimand.instrumental_variables
+        if self._instrumental_variable_names:
+            self._instrumental_variables = self._data[self._instrumental_variable_names]
+            self._instrumental_variables = pd.get_dummies(self._instrumental_variables, drop_first=True)
+        else:
+            self._instrumental_variables = []
+
+        #Check if effect modifiers are used
+        self.logger.debug("Effect Modifiers used:" + 
+                        ",".join(self._effect_modifier_names))
+        
+        
+        # Get the class corresponding the the estimator to be used
+        estimator_class = self._get_causalml_class_object(self._causalml_methodname)
+        # Initialize the object
+        # pdb.set_trace()

```

Comment:
Usually, a good practice is to remove any commented code (especially debug code) if it is not a todo for the future.